client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -931,13 +937,17 @@ class DeltaSharingRestClient(
   }
 
   private def tokenExpired(profile: DeltaSharingProfile): Boolean = {
-    if (profile.expirationTime == null) return false
-    try {
-      val expirationTime = Timestamp.valueOf(
-        LocalDateTime.parse(profile.expirationTime, ISO_DATE_TIME))
-      expirationTime.before(Timestamp.valueOf(LocalDateTime.now()))
-    } catch {
-      case _: Throwable => false
+    profile match {
+      case bt @ BearerTokenDeltaSharingProfile(_, _, _, expirationTime) =>
+        if (expirationTime == null) return false
+        try {
+          val expirationTime = Timestamp.valueOf(
+            LocalDateTime.parse(bt.expirationTime, ISO_DATE_TIME))
+          expirationTime.before(Timestamp.valueOf(LocalDateTime.now()))
+        } catch {
+          case _: Throwable => false
+        }
+      case _ => false
```

Comment:
shall we add a comment on what are other cases, and why it's always false for them.

Reply:
moved this to a method to the authProvider